File: libasynDiscordWebHook-Nukkit/src/main/java/com/ialdrich23xx/discord/WebHook.java
package com.ialdrich23xx.discord;

import com.ialdrich23xx.Loader;
import com.ialdrich23xx.discord.body.Base;
import com.ialdrich23xx.thread.SendWebHookTask;

public class WebHook {

    private String url;
    private Base body;

    public WebHook(String url, Base body) {
        this.url = url;
        this.body = body;
    }

    public static WebHook make(String url, Base body) {
        return new WebHook(url, body);
    }

    public String getUrl() {
        return this.url;
    }

    public Base getBody() {
        return this.body;
    }

    public WebHook setBody(Base body) {
        this.body = body;

        return this;
    }

    public void send() {
        if (this.getBody() == null) {
            Loader.getInstance().getLogger().error("Body of webhook is null");
            return;
        }

        if (Loader.getInstance().isValidUrl(this.getUrl())) {
            Loader.getInstance().getThreadPool().submit(new SendWebHookTask(this));
        } else Loader.getInstance().getLogger().error("Url not valid: " + this.getUrl());
    }
}

File: libasynDiscordWebHook-PM/src/ialdrich23xx/libasynwebhook/discord/WebHook.php
<?php

declare(strict_types=1);

namespace ialdrich23xx\libasynwebhook\discord;

use JsonException;
use ialdrich23xx\libasynwebhook\discord\body\Base;
use ialdrich23xx\libasynwebhook\Loader;
use ialdrich23xx\libasynwebhook\thread\SendWebHookTask;
use function is_null;

class WebHook
{
    public function __construct(
        private string $url,
        private ?Base $body = null
    ) {}

    public static function make(string $url, ?Base $body = null): self
    {
        return new self($url, $body);
    }

    public function getUrl(): string
    {
        return $this->url;
    }

    public function getBody(): ?Base
    {
        return $this->body;
    }

    public function setBody(Base $body): self
    {
        $this->body = $body;

        return $this;
    }

    /**
     * @throws JsonException
     */
    public function send(): void
    {
        if (is_null($this->getBody())) {
            Loader::getInstance()->getPlugin()->getLogger()->error("Body of webhook is null");
            return;
        }

        if (Loader::getInstance()->isValidUrl($this->getUrl())) {
            if ($this->getBody()->build()) {
                Loader::getInstance()->getThread()->submitTask(new SendWebHookTask($this));
            } else Loader::getInstance()->getPlugin()->getLogger()->error("Webhook build failed");
        } else Loader::getInstance()->getPlugin()->getLogger()->error("Url not valid: " . $this->getUrl());
    }
}